### PR TITLE
Resolve #414 Add Language Support

### LINKED ISSUE #414: Add Language Support to Backend
* Double check that survey questions/answers support unicode and not just ASCII (for supporting accented Spanish characters)
* Add language to the SurveyResponse
* Add a new text column and response_options column per language to SurveyQuestion

### PR DESCRIPTION
**Pull request by mzagaja**
* Return question text and response text in English by default, but use the headers provided by ACCEPT_LANGUAGE to pick the available language.
* Update how we import sample data.

Current sample output hitting `/surveys/1.json`

```json
{
  "id": 1,
  "title": "Draft Survey",
  "updated_at": "2023-09-26T23:42:54.764Z",
  "survey_questions": [
    {
      "id": 1,
      "display_order": 1,
      "text": "Have you heard of heat pumps?",
      "response_type": "radio",
      "response_options": [
        "Yes",
        "No"
      ]
    },
    {
      "id": 2,
      "display_order": 2,
      "text": "If yes, have you considered them for home heating/cooling?",
      "response_type": "radio",
      "response_options": [
        "Yes",
        "No"
      ]
    },
    {
      "id": 3,
      "display_order": 3,
      "text": "If yes, how far have you taken your interest in heat pumps?",
      "response_type": "text",
      "response_options": []
    },
    {
      "id": 4,
      "display_order": 4,
      "text": "Have you heard of Mass Save… the energy efficieny program?",
      "response_type": "radio",
      "response_options": [
        "Yes",
        "No"
      ]
    },
    {
      "id": 5,
      "display_order": 5,
      "text": "If yes, have you had contact with them?",
      "response_type": "radio",
      "response_options": [
        "Yes",
        "No"
      ]
    },
    {
      "id": 6,
      "display_order": 6,
      "text": "If you could save money on your utility bills with a state-discounted or free heating/cooling system upgrade, would you like to learn more?",
      "response_type": "radio",
      "response_options": [
        "Yes",
        "No"
      ]
    },
    {
      "id": 7,
      "display_order": 7,
      "text": "If yes, please confirm your name",
      "response_type": "text",
      "response_options": []
    },
    {
      "id": 8,
      "display_order": 8,
      "text": "If yes, please confirm your email address",
      "response_type": "text",
      "response_options": []
    },
    {
      "id": 9,
      "display_order": 9,
      "text": "If yes, please confirm your phone number",
      "response_type": "text",
      "response_options": []
    },
    {
      "id": 10,
      "display_order": 10,
      "text": "So we can report back on your potential savings, what type of heating system do you have? (Across the whole home, not just in individual rooms)",
      "response_type": "radio",
      "response_options": [
        "Forced hot air – vents",
        "Radiators – water",
        "Radiators – steam",
        "Basebaord – water",
        "Baseboard – electric",
        "Other"
      ]
    },
    {
      "id": 11,
      "display_order": 11,
      "text": "If other, please specify",
      "response_type": "text",
      "response_options": []
    },
    {
      "id": 12,
      "display_order": 12,
      "text": "What type of heating FUEL do you use for that heating system?",
      "response_type": "radio",
      "response_options": [
        "Oil",
        "Gas",
        "Electric (for baseboard)",
        "Other"
      ]
    },
    {
      "id": 13,
      "display_order": 13,
      "text": "If other, please specify",
      "response_type": "text",
      "response_options": []
    },
    {
      "id": 14,
      "display_order": 14,
      "text": "We expect to have “heat pump coaches” available to walk through the process with you (volunteers, who are not affiliated with any installer or brand). Would you like to know more about how to get free, personal help from a heat pump coach?",
      "response_type": "radio",
      "response_options": [
        "Yes",
        "No"
      ]
    },
    {
      "id": 15,
      "display_order": 15,
      "text": "If yes, please confirm your name",
      "response_type": "text",
      "response_options": []
    },
    {
      "id": 16,
      "display_order": 16,
      "text": "If yes, please confirm your email address",
      "response_type": "text",
      "response_options": []
    },
    {
      "id": 17,
      "display_order": 17,
      "text": "If yes, please confirm your phone number",
      "response_type": "text",
      "response_options": []
    }
  ],
  "url": "http://localhost:3000/surveys/1.json"
}
```

### CODE REVIEW

File: backend/app/models/survey_question.rb [hunk 1]
```diff
@@ -3,6 +3,20 @@
...
+  accepts_nested_attributes_for :localized_survey_questions
 
   validates :display_order, uniqueness: { scope: :survey_id }
+
+  after_initialize do
+    @language ||= 'en'
+  end
+
+  def text
+    localized_survey_questions.find_by(language_code: @language_code).text
```
Comment (mikelynch) [inline backend/app/models/survey_question.rb:16]: Should this be `@language` instead?

File: backend/app/models/survey_question.rb [hunk 2]
```diff
@@ -3,6 +3,20 @@
...
+  end
+
+  def response_options
+    localized_survey_questions.find_by(language_code: @language_code).response_options
```
Comment (mikelynch) [inline backend/app/models/survey_question.rb:20]: And similarly here.

**Review by mikelynch — APPROVED** [review]
I haven't tried running this, but it looks reasonable. Would be good to sort the CI issues but otherwise 👍

**Comment by mikelynch** [comment]
Would be good to return some sort of exception if the user requests a language code we don't have a localization for.

File: backend/app/controllers/surveys_controller.rb [hunk 3]
```diff
@@ -9,7 +10,16 @@ def index
   end
 
   # GET /surveys/1 or /surveys/1.json
-  def show; end
+  def show
+    @language_code = (params[:langPref].presence || http_accept_language.preferred_language_from(AVAILABLE_LANGUAGES))
```
Comment (mzagaja) [inline backend/app/controllers/surveys_controller.rb:14]:
```suggestion
    @language_code = (params[:langPref] if AVAILABLE_LANGUAGES.include?(params[:langPref])  || http_accept_language.preferred_language_from(AVAILABLE_LANGUAGES))
```

^ Think we need something more like this, might need to be debugged.

**Review by mzagaja — COMMENTED** [review]
Think we just need to dial-in the code for picking the language from the params and then we're good to go.

File: backend/app/controllers/surveys_controller.rb [hunk 4]
```diff
@@ -1,6 +1,7 @@
 # frozen_string_literal: true
 
 class SurveysController < ApplicationController
+  AVAILABLE_LANGUAGES = %w[en es fr-ht pt-br].freeze
```
Comment (aTegart) [inline backend/app/controllers/surveys_controller.rb:4]: Right now, the only available languages for the _survey_ are `en` and `es`.

**Review by aTegart — APPROVED** [review]

**Pull request merged**